src/commands/hld/reconcile.ts

```diff
@@ -0,0 +1,182 @@
+import commander from "commander";
+import path from "path";
+import process from "process";
+import shelljs from "shelljs";
+
+import { Bedrock } from "../../config";
+import { logger } from "../../logger";
+
+import child_process from "child_process";
+import { writeFileSync } from "fs";
+import yaml from "js-yaml";
+import { promisify } from "util";
+
+import { TraefikIngressRoute } from "../../lib/traefik/ingress-route";
+
+const exec = promisify(child_process.exec);
+
+import { IBedrockFile } from "../../types";
+
+export const reconcileHldDecorator = (command: commander.Command): void => {
+  command
+    .command("reconcile <repository-name> <hld-path>")
+    .alias("r")
+    .description("Reconcile a HLD with the services tracked in bedrock.yaml.")
+    .action(async (repositoryName, hldPath) => {
+      try {
+        if (typeof repositoryName !== "string") {
+          throw new Error(
+            `repository-name must be of type 'string', ${typeof repositoryName} given`
+          );
+        }
+
+        if (typeof hldPath !== "string") {
+          throw new Error(
+            `hld-path must be of type 'string', ${typeof hldPath} given`
+          );
+        }
+
+        const fabrikateInstalled = shelljs.which("fab");
```

Comment:
for future reference: https://github.com/microsoft/fabrikate/issues/243

Reply:
👍